Status: obsolete
# This patch should be removed after etch, as the kernel will not support
# devfs.

Index: b/fbset.c
===================================================================
--- a/fbset.c	2007-09-16 19:06:06.000000000 +0300
+++ b/fbset.c	2007-09-16 19:06:09.000000000 +0300
@@ -42,6 +42,7 @@ struct inode;
      */
 
 #define DEFAULT_FRAMEBUFFER	"/dev/fb0"
+#define DEFAULT_FRAMEBUFFER_DEVFS	"/dev/fb/0"
 
 
     /*
@@ -846,7 +847,7 @@ static void Usage(void)
 	"    -a, --all          : change all virtual consoles on this device\n"
 	"  Frame buffer special device nodes:\n"
 	"    -fb <device>       : processed frame buffer device\n"
-	"                         (default is " DEFAULT_FRAMEBUFFER ")\n"
+	"                         (default is " DEFAULT_FRAMEBUFFER "; " DEFAULT_FRAMEBUFFER_DEVFS " when using devfs)\n"
 	"  Video mode database:\n"
 	"    -db <file>         : video mode database file\n"
 	"                         (default is " DEFAULT_MODEDBFILE ")\n"
@@ -977,8 +978,12 @@ int main(int argc, char *argv[])
     if (Opt_version || Opt_verbose)
 	puts(VERSION);
 
-    if (!Opt_fb)
-	Opt_fb = DEFAULT_FRAMEBUFFER;
+    if (!Opt_fb) {
+        if (access("/dev/.devfsd", F_OK) == 0)  /* devfs detected */
+	    Opt_fb = DEFAULT_FRAMEBUFFER_DEVFS;
+	else
+	    Opt_fb = DEFAULT_FRAMEBUFFER;
+    }
 
     /*
      *  Open the Frame Buffer Device
